api/lib/infrastructure/repositories/answer-repository.js

```diff
@@ -108,16 +109,21 @@ module.exports = {
   async saveWithKnowledgeElements(answer, knowledgeElements) {
     const answerForDB = _adaptAnswerToDb(answer);
     return knex.transaction(async (trx) => {
-      const [savedAnswerDTO] = await trx('answers').insert(answerForDB).returning(COLUMNS);
-      const savedAnswer = _toDomain(savedAnswerDTO);
-      if (!_.isEmpty(knowledgeElements)) {
-        for (const knowledgeElement of knowledgeElements) {
-          knowledgeElement.answerId = savedAnswer.id;
+      const alreadySavedAnswer = await trx('answers').select('id').where({ challengeId: answer.challengeId, assessmentId: answer.assessmentId });
+      if (alreadySavedAnswer.length === 0) {
```

Comment:
Je propose de faire directement le test pour `throw` l'erreur, ce qui évite de garder l'indentation ensuite. 

```suggestion
      if (alreadySavedAnswer.length !== 0) {
      	throw new ChallengeAlreadyAnsweredError(); 
      }
     ...
```